### ISSUE TITLE
!!! Remove `'Neos.Neos:FallbackNode'` from `NodeTypeManager::getNodeType` and make `Node::nodeType` nullable

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-development-collection#4447]
The nodeTypeManager should throw if it doesnt know about the node instead of returning `'Neos.Neos:FallbackNode'`.

We often have to check for the fallback explicitly:

```php
$siteNodeType = $this->nodeTypeManager->getNodeType($nodeTypeName);

if ($siteNodeType->getName() === 'Neos.Neos:FallbackNode') {
    throw new NodeTypeNotFoundException(
        'Cannot create a site using a non-existing node type.',
        1412372375
    );
}
```

And sometime we forget about the fallback like in the command handler which currently allows non existing nodeTypeNames in a node creation command:

```php
    protected function requireNodeType(NodeTypeName $nodeTypeName): NodeType
    {
        try {
            return $this->getNodeTypeManager()->getNodeType($nodeTypeName->value);
        } catch (NodeTypeNotFoundException $exception) {
            throw new NodeTypeNotFound(
                'Node type "' . $nodeTypeName->value . '" is unknown to the node type manager.',
                1541671070
            );
        }
    }
```

**Comment by mhsdesign** [comment]
See also todo here

https://github.com/neos/neos-development-collection/blob/a301d859e065533aca7bd7ed4f5c02f763c0bf82/Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/EventSourced/Migration/NodeTypeAdjustment_Dimensions.feature#L53

@nezaniel noticed that ;)

**Comment by kitsunet** [comment]
Because the UI doesn't know what to do with it otherwise. This way you can at least select and delete or change the node.

But I agree that this might/should happen at a higher level than in the NodeTypeManager

**Comment by mhsdesign** [comment]
Also the command handler should imo not accept the fallback node type, but it does:

https://github.com/neos/neos-development-collection/blob/9d6393238062ec3eeb5e34291baa15acbb3c9cae/Neos.ContentRepository.Core/Classes/Feature/Common/ConstraintChecks.php#L103

so creating a node with invalid nodeTypeName would currently work :/

**Comment by nezaniel** [comment]
Implemented in #4466
